tool/speed.cc

```diff
@@ -1022,6 +1022,38 @@ static bool SpeedECDHCurve(const std::string &name, int nid,
   return true;
 }
 
+static bool SpeedECKeyGenCurve(const std::string &name, int nid,
```

Comment:
Calling the higher level EVP function `EVP_PKEY_keygen` ([link](https://github.com/awslabs/aws-lc/blob/8569259849f6397431e0941ec53d701fef241fed/include/openssl/evp.h#L692)) with appropriate context should take care of calling the underlying FIPS/non-FIPS implementation by itself ([link](https://github.com/awslabs/aws-lc/blob/7792c41f96d9b943cc4006ea6d7dfe733df815f5/crypto/fipsmodule/evp/p_ec.c#L228)) and you wouldn't have to worry about that here.